File: Data Structures & Algorithms/merge-intervals/submission-0.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key = lambda pair: pair[0])

        res = [intervals[0]] # we initialize the res array with a beginning []

        for start, end in intervals:
            lastEnd = res[-1][1]

            if start <= lastEnd:
                res[-1][1] = max(lastEnd, end) 
            else:
                res.append([start, end])
        return res

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0, r = 1;
        int maxProfit = 0;
        
        while ( r < prices.size()) {
            if (prices[l] < prices[r]){
                int profit = prices[r]-prices[l];
                maxProfit = max(profit, maxProfit);
            } else {
                l = r;
            }
            r++;
        }
        return maxProfit;
    }
};